internal/charm/services/repofactory.go

```diff
@@ -65,7 +62,7 @@ func (f *CharmRepoFactory) GetCharmRepository(ctx context.Context, src corecharm
 
 	switch src {
 	case corecharm.CharmHub:
-		cfg, err := f.modelBackend.Config()
+		cfg, err := f.modelConfigService.ModelConfig(ctx)
```

Comment:
We're fetching the entire model config just to cherry pick one attribute.
Wouldn't we prefer service to offer a more workflow driven api where you can ask for data that matches the use case?

Reply:
I agree with this statement. I am just worried about time constraints. I will bring it up our sync and see which way we would like to go.

Reply:
Based on last night's meeting, Ian is right but we're not going to do this now, it's too much work right now. I'll add this to the risk register and we'll agree this is fine for the current PR.